### ISSUE TITLE
In a shop, after all money is removed from pool, the Take option remains and the "Leave your stuff" message is shown

### ISSUE THREAD

**Issue by grannypron** [grannypron/uaf#35]
In a shop, if a party pools their money and then removes it all from the pool, either by Sharing or manually removing the money, the will still have the Take option and will still be presented with the message "Leave your money behind?" message when leaving the shop.

Recreated in 5.21 in TutorialDesign.dsn

**Comment by grannypron** [comment]
This was mostly a change to check and set the status of Party::moneyPooled in the cases where money was removed from the poolSack.  I tried to identify those points and stick with the convention of the one area that did this check with the line:

`      party.moneyPooled = !(party.poolSack.IsEmpty());`

HOWEVER,  It also seems that a MONEY_SACK is considered to be empty if it has no coins in it and gems/jewelry are not checked.  This would cause side effects where the player would remove all the coins and suddenly their gems and jewelry would disappear because the game considered their pool to be empty (where it never considered it to be empty before).

The code for MONEY_SACK::IsEmpty is

```
BOOL MONEY_SACK::IsEmpty()
{
  for (int i=0;i<NumCoinTypes();i++)
    if (Coins[i] > 0) return FALSE;
  return TRUE;
}
```

And I added 
```
  if (!(Gems.IsEmpty())) return FALSE;
  if (!(Jewelry.IsEmpty())) return FALSE;
```

and everything seems to work great,

However, this is a relatively low-level class and the impacts will include the treasure screen and vault screens as well as the shop screen.  I can't think of any situation where the game should count a money sack as empty if it has gems and jewelry in it, but I should check on the forums to make sure that I am not missing any nuances that I don't know about and propose this change.
